docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,926 @@
+    as we'd need some kind of dependency injection to build context before validation so models can still be passed as
+    arguments to views. I'm sure he'll be game.
+
+!!! warning
+    Although this will make it slightly easier to run synchronous IO (HTTP requests, DB. queries, etc.)
+    from within validators, I strongly advise you keep IO separate from validation - do it before and use context,
+    do it afterwards, avoid where possible making queries inside validation.
+
+### Model Namespace Cleanup :thumbsup:
+
+For years I've wanted to clean up the model namespace,
+see [pydantic#1001](https://github.com/samuelcolvin/pydantic/issues/1001). This would avoid confusing gotchas when field
+names clash with methods on a model, it would also make it safer to add more methods to a model without risking
+new clashes.
+
+After much deliberation (and even giving a lightning talk at the python language submit about alternatives, see
+[this discussion](https://discuss.python.org/t/better-fields-access-and-allowing-a-new-character-at-the-start-of-identifiers/14529)).
+I've decided to go with the simplest and clearest approach, at the expense of a bit more typing:
+
+All methods on models will start with `model_`, fields' names will not be allowed to start with `"model"`
+(aliases can be used if required).
+
+This will mean `BaseModel` will have roughly the following signature.
+
+```{.py .annotate title="New BaseModel methods"}
+class BaseModel:
+    model_fields: List[FieldInfo]
+    """previously `__fields__`, although the format will change a lot"""
+    @classmethod
+    def model_validate(cls, data: Any, *, context=None) -> Self:  # (1)
+        """
+        previously `parse_obj()`, validate data
+        """
+    @classmethod
+    def model_validate_json(
+        cls,
+        data: str | bytes | bytearray,
+        *,
+        context=None
+    ) -> Self:
+        """
+        previously `parse_raw(..., content_type='application/json')`
+        validate data from JSON
+        """
+    @classmethod
+    def model_is_instance(cls, data: Any, *, context=None) -> bool: # (2)
+        """
+        new, check if data is value for the model
+        """
+    @classmethod
+    def model_is_instance_json(
+        cls,
+        data: str | bytes | bytearray,
+        *,
+        context=None
+    ) -> bool:
+        """
+        Same as `model_is_instance`, but from JSON
+        """
+    def model_dump(
+        self,
+        include: ... = None,
+        exclude: ... = None,
+        by_alias: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        mode: Literal['unchanged', 'dicts', 'json-compliant'] = 'unchanged',
+        converter: Callable[[Any], Any] | None = None
+    ) -> Any:
+        """
+        previously `dict()`, as before
+        with new `mode` argument
+        """
+    def model_dump_json(self, ...) -> str:
```

Comment:
Like schema I would remove this personally. It helps understand that all methods that have json in them are to handle json inputs. To dump, you just need to use one json package

Reply:
Let's see, the point is we might be able to generate the JSON while converting. and therefore save a lot of time.

I'll change the comment to "equivalent to" not "alias of".